tests/integration/favorites.rs

```diff
@@ -51,57 +51,60 @@ branch = "{branch}"
 }
 
 #[test]
-fn it_survives_an_empty_config() {
-    let empty_config = tmp_dir().build();
-    let empty_config_path = empty_config.path().join("cargo-generate");
+fn favorite_with_git_becomes_subfolder() {
+    let favorite_template = create_template("favorite-template");
     let git_template = create_template("git-template");
+    let (_config, config_path) = create_favorite_config("test", &favorite_template);
     let dir = tmp_dir().build();
 
     binary()
         .arg("generate")
         .arg("--config")
-        .arg(empty_config_path)
+        .arg(config_path)
         .arg("--name")
         .arg("foobar-project")
         .arg("--git")
         .arg(git_template.path())
         .arg("test")
         .current_dir(&dir.path())
         .assert()
-        .success()
-        .stdout(predicates::str::contains("Done!").from_utf8());
-
-    assert!(dir
-        .read("foobar-project/Cargo.toml")
-        .contains(r#"description = "git-template""#));
+        .failure();
 }
 
 #[test]
-fn it_can_use_favorites() {
-    let favorite_template = create_template("favorite-template");
-    let (_config, config_path) = create_favorite_config("test", &favorite_template);
-    let dir = tmp_dir().build();
+fn favorite_with_subfolder() -> anyhow::Result<()> {
```

Comment:
question: could you also add a test for the not happy path, e.g. passing an invalid subfolder or a file to the `<subfolder>` argument?